### PR TITLE
feat: excludeFromIndex allow for '*' wildcard to catch all properties on object

### PR DESCRIPTION
**Pull request by praveenqlogic01**
towards #12 

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/request.ts [hunk 1]
```diff
@@ -1040,6 +1040,13 @@ class DatastoreRequest {
           }
         }
 
+        if (entityObject.excludeFromIndexes) {
```
Comment (stephenplusplus) [inline src/request.ts]: We have the 'excludePathFromEntity' function in entity.ts. Could this new behavior be put into the same function? This is probably the most complex bit of code in our library, so anything we can do to make it more approachable is great.

Comment (praveenqlogic01) [reply src/request.ts]: @stephenplusplus Hi, as suggested i implemented the new behavior in  function  excludePathFromEntity!

**Comment by jacekkopecky** [comment]
Thank you, this will be oh so useful!

Is there documentation anywhere?

It seems that `prop.*` excludes everything under the property `prop`; and there's a hint that a simple `.*` excludes everything. Is that correct?

If so, shouldn't that be just `*` instead? Why the initial dot?

File: system-test/datastore.ts [hunk 2]
```diff
@@ -129,6 +129,69 @@ describe('Datastore', () => {
       await datastore.delete(postKey);
     });
 
+    it('should remove index with using wildcrad in excludeFromIndexes', async () => {
```
Comment (jacekkopecky) [inline system-test/datastore.ts]: typo `wildcrat` -> `wildcard`

Comment (praveenqlogic01) [reply system-test/datastore.ts]: done!

**Comment by praveenqlogic01** [comment]
> Thank you, this will be oh so useful!
> 
> Is there documentation anywhere?
> 
> It seems that `prop.*` excludes everything under the property `prop`; and there's a hint that a simple `.*` excludes everything. Is that correct?
> 
> If so, shouldn't that be just `*` instead? Why the initial dot?

@jacekkopecky Hi, yes that's right, "." is representing root path here and * representing everything from there!

**Comment by praveenqlogic01** [comment]
verifying cla done!

**Comment by uschtwill** [comment]
This isn't documented at all, is it?

File: test/request.ts [hunk 3]
```diff
@@ -1392,6 +1392,158 @@ describe('Request', () => {
...
+          excludeFromIndexes
+        );
+        done();
+      };
+
+      request.save(
+        {
+          key,
+          data,
+          excludeFromIndexes: [
```
Comment (siddjain) [inline test/request.ts:1535]: how can i exclude all properties from being indexed? will ['*'] do it?

**Comment by kirillgroshkov** [comment]
Is there documentation for this except source code of this pull request?

**Pull request merged**